### ISSUE TITLE
Is it possible to add background images in the training set that do not have any classes?

### ISSUE THREAD

**Issue by abhisheksgumadi** [weiliu89/caffe#146]
Hi,

  Is it possible to add images that are labeled as background (the <name>background</name> node in its annotation xml file) image and annotate the whole image as a bounding box as background in the training set?

I see that after I have trained the SSD to detect a particular kind of object (object v/s background 2 classes) it is throwing a lot of false positives on human faces (that do not contain my object).

So I want SSD to consider human faces as background images. Is it possible to somehow do this?

**Comment by abhisheksgumadi** [comment]
I was thinking of it is possible to include annotation files for additional background images and not include any <object> nodes inside the annotation. In that case would the detector treat all detections as negative and thus learn to treat these additional images included in the training set as background images?

**Comment by leosteets** [comment]
I was wondering the same thing. Even with a threshold of 0.9 I'm still finding a lot of false positives.

I add some noise to the image in order to preserve the aspect ratio, I found that it increases the precision but it decreases the recall, maybe because I didn't use this technique on training, IDK. 

Anyway if it is possible to train the model providing negative samples it would be great. Please let us know if it is possible.

**Comment by weiliu89** [comment]
Currently, it only samples negatives from images which have ground truth annotation. It is possible to hack the multibox_loss_layer to also select negatives from images which do not have any ground truth (background images).

**Comment by abhisheksgumadi** [comment]
@weiliu89 can you please suggest what changes in the code is required? That would be of great help. I can make those changes, give it a try and report back my findings. Thanks.

**Comment by gurkirt** [comment]
Did anybody figured it out?

**Comment by SteveFame** [comment]
@weiliu89, could you please give us a pointer to the code where we can add this change? The multibox loss layer did change and #145 does not seem to apply anymore. Thanks.

**Comment by weiliu89** [comment]
You can search MineHardNegative in the bbox_util.cpp

**Comment by abhisheksgumadi** [comment]
Hi Wei Liu,

  Do you mean MineHardExamples function here?
https://github.com/weiliu89/caffe/blob/ssd/src/caffe/util/bbox_util.cpp#L850

Thanks
Abhishek S

Thank you!

With Regards,
Abhishek S

**Comment by weiliu89** [comment]
Yes.

**Comment by guoxiaolu** [comment]
Is it possible to add some images with 'background' label, instead purely background image without any label? @weiliu89

**Comment by sirluckyluke** [comment]
@abhisheksgumadi : i am running in similar problems like you. Did you get your solution running?

**Comment by MyVanitar** [comment]
Anybody could solve this?
